src/Steam.Common/GameId.cs

```diff
@@ -0,0 +1,188 @@
+            }
+        }
+
+        /// <summary>
+        /// The gametype of the game, see <see cref="GameTypes"/> 
+        /// </summary>
+        public GameTypes Types
+        {
+            get
+            {
+                // mask to get bits 25 - 32
+                const ushort mask = 0xFF;
+                return (GameTypes)((_value >> 24) & mask);
+            }
+        }
+        /// <summary>
+        /// Id of the mod on steam
+        /// </summary>
+        public uint ModId {
+            get
+            {
+                // mask to get the last 32 bits
+                const uint mask = 0xFFFFFFFF;
+                return (uint)((_value >> 32) & mask);
+            }
+        }
+
+        /// <summary>
+        /// Creates an instance of the GameId struct with the underlying ulong value
+        /// </summary>
+        /// <param name="value">the underlying internal value</param>
+        public GameId(ulong value)
+        {
+            _value = value;
+        }
+
+        /// <summary>
+        /// Creates an instance of the GameId struct, and converts the parameters to the underlying GameId value
+        /// </summary>
+        /// <param name="app">the app <see cref="App"/></param>
+        /// <param name="types">the type of the gameid <see cref="GameTypes"/>, defaults to <see cref="GameTypes.App"/></param>
+        /// <param name="modId">the id of the mod, defaults to 0x80000000</param>
+        public GameId(App app, GameTypes types = GameTypes.App, uint modId = 0x80000000)
+        {
+            _value = (ulong) modId << 32 | (ulong) types << 24 | app.Value;
+        }
+
+        /// <summary>
+        /// Checks if the GameId is a valid gameid
+        /// </summary>
+        /// <param name="id">the gameid to check the validity of</param>
+        /// <returns>Whether the GameId is valid</returns>
+        public static bool IsValid(GameId id)
+        {
+            // Todo: Maybe check if it is actualy an existing gameid against the steam API?
+            if (id._value == 0) return false;
+            if (!Enum.IsDefined(typeof(GameTypes), id._value << 24 & 0xFF)) return false;
+            
+            return true;
+        }
+
+        /// <summary>
+        /// Checks the gameIds for equality
+        /// </summary>
+        /// <param name="id">the other gameId</param>
+        /// <returns>Whether the GameIds are equal</returns>
+        public bool Equals(GameId id)
+        {
+            return GetHashCode() == id.GetHashCode();
+        }
+
+        /// <summary>
+        /// Checks for equality
+        /// </summary>
+        /// <param name="obj">expected GameId or ulong value</param>
+        /// <returns>whether or not the objects are equal</returns>
+        /// <exception cref="InvalidOperationException">If type can't be checked against</exception>
+        public override bool Equals(object obj)
+        {
+            if (obj is null) return false;
+            switch (obj)
```

Comment:
In this case just try and cast to a GameId and compare to keep it consistent with SteamId, App, Package, etc.